keras_cv/models/densenet.py

```diff
@@ -0,0 +1,296 @@
+    include_top,
+    classes=None,
+    weights=None,
+    input_shape=(None, None, 3),
+    pooling=None,
+    classifier_activation="softmax",
+):
+    """Instantiates the DenseNet architecture.
+
+    Reference:
+    - [Densely Connected Convolutional Networks (CVPR 2017)](
+        https://arxiv.org/abs/1608.06993)
+
+    This function returns a Keras DenseNet model.
+
+    For transfer learning use cases, make sure to read the
+    [guide to transfer learning & fine-tuning](
+      https://keras.io/guides/transfer_learning/).
+
+    Args:
+      blocks: numbers of building blocks for the four dense layers.
+      include_rescaling: whether or not to Rescale the inputs.
+        If set to True, inputs will be passed through a
+        `Rescaling(1/255.0)` layer.
+      include_top: whether to include the fully-connected
+        layer at the top of the network.  If provided, num_classes must be provided.
+      classes: optional number of classes to classify images
+        into, only to be specified if `include_top` is True, and
+        if no `weights` argument is specified.
+      weights: one of `None` (random initialization), or a pretrained
+        weight file path.
+      input_shape: optional shape tuple, defaults to (None, None, 3).
+      pooling: optional pooling mode for feature extraction
+        when `include_top` is `False`.
+        - `None` means that the output of the model will be
+            the 4D tensor output of the
+            last convolutional block.
+        - `avg` means that global average pooling
+            will be applied to the output of the
+            last convolutional block, and thus
+            the output of the model will be a 2D tensor.
+        - `max` means that global max pooling will
+            be applied.
+      classifier_activation: A `str` or callable. The activation function to use
+        on the "top" layer. Ignored unless `include_top=True`. Set
+        `classifier_activation=None` to return the logits of the "top" layer.
+        When loading pretrained weights, `classifier_activation` can only
+        be `None` or `"softmax"`.
+
+    Returns:
+      A `keras.Model` instance.
+    """
+    if not (weights in {None} or tf.io.gfile.exists(weights)):
+        raise ValueError(
+            "The `weights` argument should be either "
+            "`None` or the path to the weights file to be loaded."
+        )
+
+    if include_top and None in input_shape:
+        raise ValueError(
+            "If `include_top` is True, "
+            "you should specify a static `input_shape`. "
+            f"Received: input_shape={input_shape}"
+        )
+
+    if include_top and not classes:
+        raise ValueError(
+        "If `include_top` is True, "
+        "you should specify `classes`. "
+        f"Received: classes={classes}"
+        )
+
+    inputs = layers.Input(shape=input_shape)
+
+    x = inputs
+    if include_rescaling:
+        x = layers.Rescaling(1 / 255.0)(x)
```

Comment:
So, if we do incorporate the `mean` and `variance` arguments then the code could be amended like so:

```py
if not (mean and variance):
    if include_rescaling:
        x = layers.Rescaling(1 / 255.0)(x)
else:
    norm_layer = layers.Normalization(axis=-1, mean=mean, variance=variance)
    x = norm_layer(x)
```

I agree that rescaling to [0, 1] is way simpler and easier to do but a significant amount of models could be supported off-the-shelf with this consideration I believe.